mule-module-apikit/src/test/java/org/mule/module/apikit/parameters/ParametersNullableTestCase.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.apikit.parameters;
+
+import static com.jayway.restassured.RestAssured.given;
+import static java.lang.System.setProperty;
+
+import com.jayway.restassured.RestAssured;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+public class ParametersNullableTestCase extends FunctionalTestCase {
+
+  @Rule
+  public DynamicPort serverPort = new DynamicPort("serverPort");
+
+  @Override
+  public int getTestTimeoutSecs()
+  {
+    return 6000;
+  }
+
+  @Override
+  protected void doSetUp() throws Exception
+  {
+    setProperty("validate.null.string.param", "true");
+    RestAssured.port = serverPort.getNumber();
+    super.doSetUp();
+  }
+
+  @Override
+  protected String getConfigFile()
+  {
+    return "org/mule/module/apikit/parameters/parameters-10-config.xml";
+  }
+
+  @Test
+  public void stringTypeNullInvalid()
+  {
+    given().queryParam("string-type", (String) null)
+        .expect().response().statusCode(500)
```

Comment:
This 500s should be 400s right? should we create a separate ticket to address that ?

Reply:
This is because the example application doesnt have an error handler, if an error occurs always return 400